### PR TITLE
fix: update serialization format

### LINKED ISSUE #1166: Cannot set telemetry.tracing.otlp.grpc.metadata
**Bug Description**

This config: 
```yaml
telemetry:
  tracing:
    otlp:
      endpoint: otlp.nr-data.net:4317
      grpc:
        metadata:
          "api-key": ${NEW_RELIC_LICENSE_KEY}
```
causes this error at startup:
```
2022-05-26T20:39:30.605914Z ERROR apollo_router::router_factory: plugin apollo.telemetry could not be configured: invalid type: map, expected a sequence
2022-05-26T20:39:30.606164Z ERROR apollo_router::state_machine: cannot create the router: plugin apollo.telemetry could not be configured: invalid type: map, expected a sequence
2022-05-26T20:39:30.606245Z DEBUG apollo_router::state_machine: transitioned to state errored
2022-05-26T20:39:30.606255Z DEBUG apollo_router::state_machine: stopped
2022-05-26T20:39:30.606331Z  INFO apollo_router: stopped with error
2022-05-26T20:39:30.606508Z ERROR apollo_router::executable: could not create the HTTP pipeline: plugin apollo.telemetry could not be configured: invalid type: map, expected a sequence
Error: could not create the HTTP pipeline: plugin apollo.telemetry could not be configured: invalid type: map, expected a sequence
```

I couldn't figure out how to send a "sequence". I also tried `metadata: {}` and `metadata: "api-key: ${NEW_RELIC_LICENSE_KEY}` and got the same error.

**Versions**
 - router 0.9.2

**Additional context**
I was able to send traces to newrelic by using a collector, so this isn't a high-priority bug.

### PR DESCRIPTION
**Pull request by me-diru**
- fixes #1166
- changes data structure to a `IndexMap<String, Vec < String>>`
- updated test case

Signed-off-by: Rohit Dandamudi <rohitdandamudi.1100@gmail.com>

### CODE REVIEW

**Comment by me-diru** [comment]
@bnjjj I am not sure about this, the `HashMap<String, String>` type doesn't allow duplicate key entries(which is what we want I think?), and also it doesn't maintain any order which leads to failing tests when I tried. 

I didn't update CHANGELOG yet ( I think we should update the NEXT_CHANGELOG though?).

let me know your thoughts, thanks!

**Comment by bnjjj** [comment]
SOrry for the delay @me-diru I was on vacation. I checked the JavaScript implementation to know more about the GRPC metadata and discover that https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/metadata.ts#L25. So I think the format we want is `HashMap<String, Vec<String>>` and we don't care about ordering. You should change the type [here](https://github.com/apollographql/router/blob/cf422d04aeff604732eacab8494207673eae4e06/apollo-router/src/plugins/telemetry/otlp.rs#L206) and [here](https://github.com/apollographql/router/blob/cf422d04aeff604732eacab8494207673eae4e06/apollo-router/src/plugins/telemetry/otlp.rs#L224). Does it help you ?

**Comment by me-diru** [comment]
Thanks for the response @bnjjj, I tried to follow that approach. Let me know your thoughts, thanks!

**Comment by me-diru** [comment]
It still doesn't get the order of keys right, resulting in failing tests. Hashmap doesn't store data in any order, there's no static case to check. 

Avoiding Vector comes at a cost of not maintaining the order of key storage.

File: apollo-router/src/plugins/telemetry/otlp.rs [hunk 1]
```diff
@@ -233,11 +247,10 @@ mod metadata_map_serde {
...
 
-        for submap in serializable_format.into_iter() {
-            for (key, value) in submap.into_iter() {
-                let key = MetadataKey::from_bytes(key.as_bytes()).unwrap();
-                map.append(key, value.parse().unwrap());
-            }
+        for (key, values) in serializable_format.into_iter() {
+            let key = MetadataKey::from_bytes(key.as_bytes()).unwrap();
+            let last_value = values.last().unwrap();
+            map.append(key, last_value.parse().unwrap());
```
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/otlp.rs:253]: Do not take only the last value but append all values

File: apollo-router/src/plugins/telemetry/snapshots/apollo_router__plugins__telemetry__otlp__metadata_map_serde__tests__serialize_metadata_map.snap [hunk 2]
```diff
@@ -3,7 +3,8 @@ source: apollo-router/src/plugins/telemetry/otlp.rs
 expression: "std::str::from_utf8(&buffer).unwrap()"
 ---
 ---
-- foo: bar
-- foo: baz
-- bar: foo
-
+foo:
```
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/snapshots/apollo_router__plugins__telemetry__otlp__metadata_map_serde__tests__serialize_metadata_map.snap:6]: If you want to keep ordering to not break the test then use `IndexMap` from https://docs.rs/indexmap/latest/indexmap/map/struct.IndexMap.html it should already exist in our deps

**Review by bnjjj — CHANGES_REQUESTED** [review]
Great job, 2 last bits and I think we're good. Please ensure to properly fill the `NEXT_CHANGELOG.md` file with your changes

**Review by bnjjj — APPROVED** [review]

**Comment by bnjjj** [comment]
Great job @me-diru thanks for your contribution !

**Pull request merged**